src/jdk.incubator.vector/share/classes/jdk/incubator/vector/ByteVector.java

```diff
@@ -4101,7 +4101,7 @@ static long booleanArrayAddress(boolean[] a, int index) {
 
     @ForceInline
     static long byteArrayAddress(byte[] a, int index) {
-        return Unsafe.ARRAY_BYTE_BASE_OFFSET + index;
+        return (long) Unsafe.ARRAY_BYTE_BASE_OFFSET + index;
```

Comment:
These classes are generated via the `X-Vector.java.template` in the `java.incubator.vector` package.  So, you only need to change one file here.

![image](https://github.com/user-attachments/assets/2dbda48c-ae17-4809-a46f-62065ee9eab3)